gammapy/data/pointing.py

```diff
@@ -125,40 +149,40 @@ def __init__(
             self.__dict__.update(self.from_fits_header(meta).__dict__)
             return
 
-        if not isinstance(mode, PointingMode):
-            raise TypeError(f"mode must be an instance of PointingMode, got {mode!r}")
+        if mode is not None:
+            warnings.warn(
+                "Passing mode is deprecated and the argument will be removed in Gammapy 2.0."
+                " pointing mode is deduced from whether fixed_icrs or fixed_altaz is given",
+                GammapyDeprecationWarning,
+            )
 
-        self._mode = mode
         self._location = location
         self._time_start = time_start
         self._time_stop = time_stop
         self._time_ref = time_ref
         self._legacy_altaz = legacy_altaz or AltAz(np.nan * u.deg, np.nan * u.deg)
 
-        if mode is PointingMode.POINTING:
-            if fixed_icrs is None:
-                raise ValueError("fixed_icrs is mandatory for PointingMode.POINTING")
+        if fixed_icrs is not None and fixed_altaz is not None:
+            raise ValueError("fixed_icrs and fixed_altaz are mutually exclusive")
+
+        if fixed_icrs is not None:
+            _check_coord_frame(fixed_icrs, ICRS, "fixed_icrs")
 
             if np.isnan(fixed_icrs.ra.value) or np.isnan(fixed_icrs.dec.value):
                 warnings.warn(
                     "In future, fixed_icrs must have non-nan values",
                     GammapyDeprecationWarning,
                 )
 
+            self._mode = PointingMode.POINTING
             self._fixed_icrs = fixed_icrs
             self._fixed_altaz = None
 
-        elif mode is PointingMode.DRIFT:
-            if fixed_altaz is None:
-                raise ValueError("pointing_altaz is required for PointingMode.DRIFT")
-
-            if fixed_icrs is not None:
-                raise ValueError("fixed_icrs is excluded for PointingMode.DRIFT")
-
-            self._fixed_altaz = fixed_altaz
-            self._fixed_icrs = None
         else:
-            raise ValueError(f"Unsupported pointing mode for FixedPointingInfo: {mode}")
+            _check_coord_frame(fixed_altaz, AltAz, "fixed_altaz")
+            self._mode = PointingMode.DRIF
+            self._fixed_icrs = None
+            self._fixed_altaz = fixed_altaz
```

Comment:
It makes the code in the getters much easier and it doens't really cost much, right? Instead of checking in the getters the frame, these can just return the internal state.